src/hypothesis/stateful.py

```diff
@@ -354,13 +353,59 @@ def precondition_wrapper(*args, **kwargs):
             setattr(precondition_wrapper, PRECONDITION_MARKER, precond)
         else:
             new_rule = Rule(targets=rule.targets, arguments=rule.arguments,
-                            function=rule.function, precondition=precond,
-                            parent_rule=rule.parent_rule)
+                            function=rule.function, precondition=precond)
             setattr(precondition_wrapper, RULE_MARKER, new_rule)
+
+        invariant = getattr(f, INVARIANT_MARKER, None)
+        if invariant is not None:
+            new_invariant = Invariant(function=invariant.function,
+                                      precondition=precond)
+            setattr(precondition_wrapper, INVARIANT_MARKER, new_invariant)
+
         return precondition_wrapper
     return decorator
 
 
+Invariant = namedtuple(
+    u'Invariant',
```

Comment:
I know you've copied this from the other definitions (which are the result of some automated refactoring when I removed a "from \_\_future\_\_ import unicode_literals" from the project that was a bit overzealous), but these should actually be bare strings and not unicode strings and I just haven't fixed all the places where this isn't quite right.